src/OSPSuite.Presentation/PresentationRegister.cs

```diff
@@ -84,6 +80,10 @@ public override void RegisterInContainer(IContainer container)
          container.Register<ChartEditorAndDisplaySettings, ChartEditorAndDisplaySettings>();
          container.Register<ChartEditorSettings, ChartEditorSettings>();
          container.Register<ChartPresenterContext, ChartPresenterContext>();
+         // These registrations need to be provided since
+         // the Factory needs the actual Implementation now, not the Interface.
+         container.Register<HeavyWorkPresenter, HeavyWorkPresenter>();
+         container.Register<HeavyWorkCancellablePresenter, HeavyWorkCancellablePresenter>();
```

Comment:
If the factory does it's work using interface, then it should be fine to not register explicitly and instead rely on convention

```
         if (supportsCancellation)
            return Create<IHeavyWorkCancellablePresenter>();  
         return Create<IHeavyWorkPresenter>();  
```